lib/zipkin-tracer/faraday/zipkin-tracer.rb

```diff
@@ -17,6 +17,7 @@ def call(env)
         b3_headers.each do |method, header|
           env[:request_headers][header] = trace_id.send(method).to_s
         end
+        env[:request_headers]['X-B3-Sampled'] = env[:request_headers]['X-B3-Sampled'] == 'true' ? '1' : '0'
```

Comment:
This probably should be in the rack middleware instead if I understand what you are doing.
Aren't you trying to get all incoming `true` to become `1`?